pkg/f1/run/test_runner.go

```diff
@@ -187,28 +187,38 @@ func (r *Run) run() {
 	r.result.RecordStarted()
 	defer r.result.RecordTestFinished()
 
-	doWork := make(chan bool, workers)
+	workTriggered := make(chan bool, workers)
 	stopTrigger := make(chan bool, 1)
-	go r.trigger.Trigger(doWork, stopTrigger, workDone, r.Options)
+	go r.trigger.Trigger(workTriggered, stopTrigger, workDone, r.Options)
+
+	// This blocks waiting for cancellable timer
+	go func() {
+		<-durationElapsed.C
+		trace.ReceivedFromChannel("C")
+		fmt.Println(r.result.MaxDurationElapsed())
+		log.Info("Stopping worker")
+		stopTrigger <- true
+		close(stopWorkers)
+		wg.Wait()
```

Comment:
This prints out max durations elapsed if it stops for any reason (e.g. interrupt / max iterations). Not tested, but think we need something like this instead
 ```suggestion
		_, ok := <-durationElapsed.C
		trace.ReceivedFromChannel("C")
		if ok {
		  fmt.Println(r.result.MaxDurationElapsed())  
		}
		log.Info("Stopping worker")
		stopTrigger <- true
		close(stopWorkers)
		wg.Wait()
```

Reply:
Ah yes, good point, I had noticed that - I'll take a look :+1:

Reply:
Fixed in ad24e3c8f258824ea21b1b7e2c1553954b84d891